modules/gwlb/main.tf

```diff
@@ -1,61 +1,64 @@
-    private_ip_address_allocation = try(var.frontend_ip_config.private_ip_address_allocation, null)
-    private_ip_address_version    = try(var.frontend_ip_config.private_ip_address_version, null)
-    private_ip_address            = try(var.frontend_ip_config.private_ip_address, null)
-    subnet_id                     = var.frontend_ip_config.subnet_id
-    zones                         = try(var.frontend_ip_config.zones, null)
+    name                          = var.frontend_ip.name
+    private_ip_address_allocation = var.frontend_ip.private_ip_address != null ? "Static" : "Dynamic"
+    private_ip_address_version    = var.frontend_ip.private_ip_address_version
+    private_ip_address            = var.frontend_ip.private_ip_address
+    subnet_id                     = var.frontend_ip.subnet_id
+    zones                         = var.frontend_ip.subnet_id != null ? var.zones : null
   }
-
-  tags = var.tags
 }
 
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_backend_address_pool
 resource "azurerm_lb_backend_address_pool" "this" {
   for_each = var.backends
 
-  name            = try(each.value.name, "${var.name}-${each.key}")
+  name            = coalesce(each.value.name, "${var.name}-${each.key}")
   loadbalancer_id = azurerm_lb.this.id
 
   dynamic "tunnel_interface" {
     for_each = each.value.tunnel_interfaces
     content {
       identifier = tunnel_interface.value.identifier
       port       = tunnel_interface.value.port
-      protocol   = "VXLAN"
+      protocol   = tunnel_interface.value.protocol
       type       = tunnel_interface.value.type
     }
   }
 }
 
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_probe
 resource "azurerm_lb_probe" "this" {
-  name            = try(var.health_probe.name, var.name)
+  name            = coalesce(var.health_probe.name, var.name)
   loadbalancer_id = azurerm_lb.this.id
 
-  port                = try(var.health_probe.port, null)
-  protocol            = try(var.health_probe.protocol, null)
-  probe_threshold     = try(var.health_probe.probe_threshold, null)
-  request_path        = try(var.health_probe.request_path, null)
-  interval_in_seconds = try(var.health_probe.interval_in_seconds, null)
-  number_of_probes    = try(var.health_probe.number_of_probes, null)
+  port                = var.health_probe.port
+  protocol            = var.health_probe.protocol
+  probe_threshold     = var.health_probe.probe_threshold
+  request_path        = var.health_probe.request_path
+  interval_in_seconds = var.health_probe.interval_in_seconds
+  number_of_probes    = var.health_probe.number_of_probes
 }
 
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_rule
 resource "azurerm_lb_rule" "this" {
-  name            = try(var.lb_rule.name, azurerm_lb.this.frontend_ip_configuration[0].name)
+  name            = coalesce(var.lb_rule.name, azurerm_lb.this.frontend_ip_configuration[0].name)
```

Comment:
Ahh, now I see why this property was optional inside the variables file. Such name generation probably complies with our approach, perhaps to be discussed?

Reply:
Yes, let's discuss it.

Reply:
I've made `name` as required, name generation logic is moved to the example.